### PR TITLE
[expo] stop exporting LegacyNotifications in SDK 41

### PR DESCRIPTION
**Pull request by cruzach**
# Why

This module has been deprecated for the past 2 SDK releases, we said we'd remove it in SDK 40 but didn't so let's remove it now

# How

Removed export, and changed some logic in `ExpoFirebaseMessagingDelegate.java` (which was already preventing any remote notifications from using the legacy API in SDK 40 😬 )

# Revert plan

Are you trying to revert this PR after releasing SDK 41 because you want to re-enable the legacy notifications module? Since we were preventing remote notifications from using the legacy API in SDK 40, it's not as simple as just rolling back this PR, but it is _almost_ that simple. Revert this PR, then apply this diff to `ExpoFirebaseMessagingDelegate.java`:
```diff
--- a/android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.java
+++ b/android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.java
@@ -53,6 +53,16 @@ public class ExpoFirebaseMessagingDelegate extends FirebaseMessagingDelegate {
         // If an experience is on SDK newer than 39, that is SDK40 and beyond up till UNVERSIONED
         // we only use the new notifications API as it is going to be removed from SDK40.
         if (sdkVersion >= 40) {
+          // BEGIN- IN CASE OF EMERGENCY
+          JSONObject androidSection = manifest.optJSONObject("android");
+          if (androidSection != null) {
+          boolean useNextNotificationsApi = androidSection.optBoolean("useNextNotificationsApi", true);
+            if (!useNextNotificationsApi) {
+              dispatchToLegacyNotificationModule(remoteMessage);
+              return;
+            }
+          }
+          // END- IN CASE OF EMERGENCY
           dispatchToNextNotificationModule(remoteMessage);
           return;
         } else if (sdkVersion == 38 || sdkVersion == 39) {
```
 and instruct users to set `android.useNextNotificationsApi` in their app.json file to `false`. Then republish the `expo` package and rebuild android shell apps.

### CODE REVIEW

File: docs/scripts/schemas/unversioned/app-config-schema.js [hunk 1]
```diff
@@ -665,7 +665,7 @@ export default {
       },
       useNextNotificationsApi: {
         description:
-          'A Boolean value that indicates whether the app should use the new notifications API.',
+          'A Boolean value that indicates whether the app should use the new notifications API. Only applies to SDK 40 and below- in SDK 41, you must use the new API.',
```
Comment (ide) [inline docs/scripts/schemas/unversioned/app-config-schema.js:668]:
```suggestion
          'A Boolean value that indicates whether the app should use the modern notifications API. Only applies to SDK 40 and below - SDK 41 and later do not include the legacy notifications API.',
```

**Review by ide — APPROVED** [review]
Rename apps/test-suite/tests/NewNotifications.ts to apps/test-suite/tests/Notifications.ts.

**Comment by brentvatne** [comment]
I think we can remove `badgin` now as well:
https://github.com/expo/expo/blob/master/packages/expo/package.json#L58

File: android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.java [hunk 2]
```diff
@@ -50,13 +50,13 @@ public void onMessageReceived(@NonNull RemoteMessage remoteMessage) {
         JSONObject manifest = new JSONObject(experienceDBObject.manifest);
         int sdkVersion = ABIVersion.toNumber(manifest.getString("sdkVersion")) / 10000;
 
-        // If an experience is on SDK newer than 39, that is SDK40 and beyond up till UNVERSIONED
-        // we only use the new notifications API as it is going to be removed from SDK40.
-        if (sdkVersion >= 40) {
+        // If an experience is on SDK 41 or above, use the new notifications API
+        // It is removed beginning with SDK41
+        if (sdkVersion > 40) {
```
Comment (brentvatne) [inline android/expoview/src/main/java/host/exp/exponent/fcm/ExpoFirebaseMessagingDelegate.java:55]:
```suggestion
        if (sdkVersion >= 41) {
```

this matches more closely with the comment and imo is slightly easier to grok at a glance

File: docs/scripts/schemas/unversioned/app-config-schema.js (hunk 1 above)
Comment (brentvatne) [reply docs/scripts/schemas/unversioned/app-config-schema.js:668]: we should mark this as deprecated

File: packages/expo/src/launch/withExpoRoot.tsx [hunk 3]
```diff
@@ -22,10 +21,6 @@ export default function withExpoRoot<P extends InitialProps>(
   return function ExpoRoot(props: P) {
     const didInitialize = React.useRef(false);
     if (!didInitialize.current) {
-      if (props.exp?.notification) {
```
Comment (brentvatne) [inline packages/expo/src/launch/withExpoRoot.tsx:25]: it looks like we can get rid of the didInitialize ref entirely

**Review by brentvatne — CHANGES_REQUESTED** [review]
just a few small things! i'm very pleased that we're finally able to do this. thanks!

**Review by brentvatne — APPROVED** [review]

**Pull request merged**